clio-server/src/main/scala/org/broadinstitute/clio/server/webservice/IndexWebService.scala

```diff
@@ -0,0 +1,69 @@
+package org.broadinstitute.clio.server.webservice
+
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.server.Directives._
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.{
+  ClioDocument,
+  ElasticsearchIndex
+}
+import org.broadinstitute.clio.server.service.IndexService
+import org.broadinstitute.clio.transfer.model.TransferIndex
+
+abstract class IndexWebService[TI <: TransferIndex, D <: ClioDocument: ElasticsearchIndex](
+  val indexService: IndexService[TI, D]
+) extends JsonWebService {
+
+  import indexService.transferIndex.implicits._
+
+  val servicePathPrefix: String
+
+  lazy val routes: Route = {
+    pathPrefix("v1") {
+      pathPrefix(servicePathPrefix) {
+        concat(getSchema, postMetadata, query, queryall)
+      }
+    }
+  }
+
+  private[webservice] val pathPrefixKey: Directive1[indexService.transferIndex.KeyType]
```

Comment:
Can you make this a `def`? It's general Scala best-practice, because it lets the individual implementations choose whether the implementation should use a `def`, `lazy val`, or `val`.